azurelinuxagent/common/cgroupapi.py

```diff
@@ -497,37 +497,32 @@ def create_cgroup(controller):
         self.track_cgroups(extension_cgroups)
 
         # Wait for process completion or timeout
-        timed_out, return_code = wait_for_process_completion_or_timeout(process, timeout)
-        process_output = read_output(stdout, stderr)
-
-        if return_code == 0:
-            # The process terminated in time and successfully
-            return extension_cgroups, process_output
-        else:
+        try:
+            process_output = handle_process_completion(process=process,
+                                                       command=command,
+                                                       timeout=timeout,
+                                                       stdout=stdout,
+                                                       stderr=stderr,
+                                                       error_code=error_code)
+        except ExtensionError as e:
             # The extension didn't terminate successfully. Determine whether it was due to systemd errors or
             # extension errors.
+            process_output = read_output(stdout, stderr)
             systemd_failure = self._is_systemd_failure(scope_name, process_output)
 
             if not systemd_failure:
-                # There was an extension error; it either timed out or returned a non-zero exit code.
-                if timed_out:
-                    raise ExtensionError("Timeout({0}): {1}\n{2}".format(timeout, command, process_output),
-                                         code=ExtensionErrorCodes.PluginHandlerScriptTimedout)
-                else:
-                    raise ExtensionError("Non-zero exit code: {0}, {1}\n{2}".format(return_code,
-                                                                                    command,
-                                                                                    process_output),
-                                         code=error_code)
+                # There was an extension error; it either timed out or returned a non-zero exit code. Re-raise the error
+                raise
             else:
                 # There was an issue with systemd-run. We need to log it and retry the extension without systemd.
+                err_msg = 'Systemd process exited with code %s and output %s' % (e.exit_code, process_output) \
+                    if isinstance(e, ExtensionOperationError) else "Systemd Timed-out, output: %s" % process_output
```

Comment:
nit: unnecessary capitalization in Timed-out

Reply:
corrected :)